crates/wasmtime/src/store.rs

```diff
@@ -745,11 +745,25 @@ impl<T> Store<T> {
     /// If fuel consumption is not enabled via
     /// [`Config::consume_fuel`](crate::Config::consume_fuel) then this
     /// function will return `None`. Also note that fuel, if enabled, must be
-    /// originally configured via [`Store::add_fuel`].
+    /// originally configured via [`Store::add_fuel`] or [`Store::set_fuel`].
+    ///
+    /// Note that this only returns the amount of fuel consumed since the last
+    /// time [`set_fuel`][Store::set_fuel] was called, or since the creation
```

Comment:
Or `add_fuel`, right?

Reply:
Ach, sorry this took so long to respond to; but no, add_fuel doesn't reset the fuel counter, that's why `fuel_adj` exists. add_fuel does like `fuel += arg; fuel_adj += arg;` so that `fuel_consumed` can return `fuel_adj - fuel`. There was a discussion somewhere I saw about concerns with fuel_adj overflowing and personally I'm skeptical of the use-case for fuel_consumed tracking across multiple add_fuels (maybe that's what the c-api requires?), but it'd be a breaking change to change that and I just wanted to provide an alternative with this.